source/dotnet/Library/AdaptiveCards/AdaptiveTypedElementConverter.cs

```diff
@@ -66,84 +55,115 @@ public static void RegisterTypedElement<T>(string typeName = null)
-                    if (Ids.Contains(objectId))
-                    {
-                        throw new AdaptiveSerializationException($"Duplicate 'id' found: '{objectId}'");
-                    }
-                    else
-                    {
-                        Ids.Add(objectId);
-                    }
+                    internalID = AdaptiveInternalID.Next();
+                    ParseContext.PushElement(objectId, internalID);
                 }
 
                 var result = (AdaptiveTypedElement)Activator.CreateInstance(type);
                 try
                 {
                     serializer.Populate(jObject.CreateReader(), result);
                 }
-                catch (JsonSerializationException)
+                catch (JsonSerializationException) { }
+
+                // remove id of element from ParseContext
+                if (type != typeof(AdaptiveCard))
+                    ParseContext.PopElement();
+
+                HandleAdditionalProperties(result);
+                return result;
+            }
+            else // We're looking at an unknown element
+            {
+                string objectId = jObject.Value<string>("id");
+                AdaptiveInternalID internalID = AdaptiveInternalID.Next();
+
+                // Handle deserializing unknown element
+                ParseContext.PushElement(objectId, internalID);
+                AdaptiveTypedElement result = null;
+                if (ParseContext.Type == ParseContext.ContextType.Element)
+                {
+                    try
+                    {
+                        result = (AdaptiveTypedElement)Activator.CreateInstance(typeof(AdaptiveUnknownElement));
+                        serializer.Populate(jObject.CreateReader(), result);
+                    }
+                    catch (JsonSerializationException) { }
+                }
+                else // ParseContext.Type == ParseContext.ContextType.Action
                 {
-                    return result;
+                    try
+                    {
+                        result = (AdaptiveTypedElement)Activator.CreateInstance(typeof(AdaptiveUnknownAction));
+                        serializer.Populate(jObject.CreateReader(), result);
+                    }
+                    catch (JsonSerializationException) { }
                 }
+                ParseContext.PopElement();
 
-                HandleAdditionalProperties(result);
+                Warnings.Add(new AdaptiveWarning(-1, $"Unknown element '{typeName}'"));
                 return result;
             }
+        }
 
-            Warnings.Add(new AdaptiveWarning(-1, $"Unknown element '{typeName}'"));
-            return null;
+        public static string GetElementTypeName(Type objectType, JObject jObject)
+        {
+            var typeName = jObject["type"]?.Value<string>() ?? jObject["@type"]?.Value<string>();
+            if (typeName == null)
```

Comment:
>ypeName == null [](start = 17, length = 15)

Nit: !typeName

Reply:
doesn't compile :(

---
In reply to: [278324249](https://github.com/Microsoft/AdaptiveCards/pull/2560#discussion_r278324249) [](ancestors = 278324249)

Reply:
HMm i wonder if it's because we are using a older framework version that didnt have support for isNullOrWhiteSpace()

---
In reply to: [278346453](https://github.com/Microsoft/AdaptiveCards/pull/2560#discussion_r278346453) [](ancestors = 278346453,278324249)